run.py

```diff
@@ -916,33 +968,32 @@ def main():
 
     # Sort alphabetically.
     tests.sort()
-    failedLogs = []
 
-    for dirpath in tests:
-        name = os.path.basename(dirpath)
+    if LINUX:
+        jobs = args.j
+        print("Number of concurrent jobs: %d" % jobs)
+
+        pool = mp.Pool(jobs)
+    
+        try:
+            for dirpath in tests:
+                pool.apply_async(run_test, args=(dirpath, args, cwd, suricata_config))
+        except TerminatePoolError:
+            pool.terminate()
 
-        outdir = os.path.join(dirpath, "output")
-        if args.outdir:
-            outdir = os.path.join(os.path.realpath(args.outdir), name, "output")
+        pool.close()
+        pool.join()
 
-        test_runner = TestRunner(
-            cwd, dirpath, outdir, suricata_config, args.verbose, args.force)
+    else:
         try:
-            results = test_runner.run()
-            if results["failure"] > 0:
-                failed += 1
-                failedLogs.append(dirpath)
-            elif results["skipped"] > 0 and results["success"] == 0:
-                skipped += 1
-            elif results["success"] > 0:
-                passed += 1
-        except UnsatisfiedRequirementError as ue:
-            print("SKIPPED: {}".format(ue))
-            skipped += 1
-        except TestError as te:
-            print("FAILED: {}".format(te))
-            check_args_fail()
-            failed += 1
+            for dirpath in tests:
+                run_test(dirpath, args, cwd, suricata_config)
+        except TerminatePoolError:
+            sys.exit(1)
```

Comment:
Can we break this into 2 functions...  `run_mp` and `run_single` or something like that?  Will just add a bit a tidy to the main method and then I think we're good for now.  Time depending we can come back around on threading/mac/windows.